ViewControllers/DetailViewController.swift

```diff
@@ -203,29 +238,62 @@ class DetailViewController: UIViewController {
 
-            languageStack.leftAnchor.constraint(equalTo: releaseStack.leftAnchor),
-            languageStack.topAnchor.constraint(equalTo: movieDescriptionLabel.bottomAnchor, constant: 16),
-
-            genreStack.leftAnchor.constraint(equalTo: releaseStack.leftAnchor),
-            genreStack.topAnchor.constraint(equalTo: languageStack.bottomAnchor, constant: 8)
+            collectionView.leftAnchor.constraint(equalTo: releaseStack.leftAnchor),
+            collectionView.topAnchor.constraint(equalTo: movieDescriptionLabel.bottomAnchor, constant: 16),
+            collectionView.rightAnchor.constraint(equalTo: movieDescriptionLabel.rightAnchor),
+            collectionView.heightAnchor.constraint(equalToConstant: 200),
         ])
     }
     
-    func configureWithData() {
+    private func configureWithData() {
         guard let movieDetails = movieDetails else { return }
 
         titleLabel.text = movieDetails.title
         
-        let path = movieDetails.posterPath
-            if let url = movieDataService.getMoviePosterURL(posterPath: path) {
-                imageView.kf.setImage(with: url)
-            }
+        if let path = movieDetails.posterPath {
+            imageView.kf.setImage(with: path)
+        }
         
-        releaseDateLabel.text = movieDetails.releaseDate
-        durationLabel.text = "\(movieDetails.runtime)m"
+        releaseDateLabel.text = "\(movieDetails.releaseDate)"
+        durationLabel.text = movieDetails.runtime
         movieDescriptionLabel.text = movieDetails.overview
         languages = movieDetails.spokenLanguages.map{$0.englishName}
         genres = movieDetails.genres.map{$0.name}
     }
 
 }
+
+extension DetailViewController: UICollectionViewDataSource {
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return viewData.count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        assert(!languages.isEmpty, "Languages array is empty")
+        return viewData[section].items.count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        
+        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: DetailCollectionViewCell.reuseIdentifier, for: indexPath) as? DetailCollectionViewCell else { fatalError() }
+        let item = viewData[indexPath.section].items[indexPath.item].title.lowercased().capitalizingFirstLetter()
+        
+        if viewData[indexPath.section].identifier == .languages {
+        cell.configure(title: item, backgroundColor: UIColor(named: "languageBackgroundColor")!, borderColor: UIColor(named: "languageBorderColor")!)
+        } else {
+            cell.configure(title: item, backgroundColor: UIColor(named: "genreBackgroundColor")!, borderColor: UIColor(named: "genreBorderColor")!)
+        }
```

Comment:
Just a small suggestion, in cases like this I feel that `switch` make it more readable so it would be:
```suggestion
        switch viewData[indexPath.section].identifier {
        case .languages:
            cell.configure(title: item, backgroundColor: UIColor(named: "languageBackgroundColor")!, borderColor: UIColor(named: "languageBorderColor")!)
        
        case .genres:
             cell.configure(title: item, backgroundColor: UIColor(named: "genreBackgroundColor")!, borderColor: UIColor(named: "genreBorderColor")!)
        }
```
Then, in case you add more sections the compiler will remind you to include the config as well